tests/Finder/TemplateFinderTest.php

```diff
@@ -33,9 +33,45 @@ public function testFindOne()
         $this->assertSame('custom.mjml.twig', $template->getFilename());
     }
 
+    public function testFindTwo()
+    {
+        $finder = new TemplateFinder(__DIR__ . '/../Functional', [
+            '/templates/mjml/',
+            '/templates/folder/subfolder/mjml/'
+        ]);
+
+        $templates = $finder->find('custom.mjml.twig');
+
+        //first template
+        $this->assertCount(1, $templates);
+
+        $template = array_pop($templates);
+        $this->assertSame('custom.mjml.twig', $template->getFilename());
+
+        //second template
+        $templates = $finder->find('custom_path.mjml.twig');
+
+        $this->assertCount(1, $templates);
+
+        $template = array_pop($templates);
+        $this->assertSame('custom_path.mjml.twig', $template->getFilename());
+    }
+
+    public function testFindDuplicates()
+    {
+        $finder = new TemplateFinder(__DIR__ . '/../Functional', [
+            '/templates/mjml/',
+            '/templates/folder/subfolder/mjml/',
+            '/templates/folderWithDuplicates/subfolder/mjml/'
+        ]);
+
+        $this->expectExceptionMessage('Duplicates template name found custom_path.mjml.twig');
```

Comment:
You should also test for the exception class

Reply:
done